packages/credential-eip712/src/agent/CredentialEIP712.ts

```diff
@@ -124,44 +116,191 @@ export class CredentialIssuerEIP712 implements IAgentPlugin {
 
   /** {@inheritdoc ICredentialIssuerEIP712.verifyCredentialEIP712} */
   private async verifyCredentialEIP712(args: IVerifyCredentialEIP712Args, context: IRequiredContext): Promise<boolean> {
+    const { credential } = args
+    if (!credential.proof || !credential.proof.proofValue) throw new Error("Proof is undefined")
+    if (
+      !credential.proof.eip712 ||
+      !credential.proof.eip712.messageSchema ||
+      !credential.proof.eip712.domain
+    ) throw new Error("eip712 is undefined");
+
+    const { proof, ...signingInput } = credential;
+    const { proofValue, eip712, ...verifyInputProof } = proof;
+    const verificationMessage = {
+      ...signingInput,
+      proof: verifyInputProof
+    }
+
+    const objectToVerify = {
+      message: verificationMessage,
+      domain: eip712.domain,
+      types: eip712.messageSchema,
+      primaryType: eip712.primaryType
+    }
+
+    const recovered = recoverTypedSignature({
+      data: objectToVerify,
+      signature: proofValue,
+      version: SignTypedDataVersion.V4
+    })
+
+    const issuer = extractIssuer(credential)
+    if (!issuer || typeof issuer === 'undefined') {
+      throw new Error('invalid_argument: args.credential.issuer must not be empty')
+    }
+
+    const didDocument = await resolveDidOrThrow(issuer, context);
+
+    if (didDocument.verificationMethod) {
+      for (const verificationMethod of didDocument.verificationMethod) {
+        if (getEthereumAddress(verificationMethod)?.toLowerCase() === recovered.toLowerCase()) {
+          return true
+        }
+      }
+    }
+    else {
+      throw new Error("Recovered Address does not match issuer")
+    }
+
+    if (await isRevoked(credential, context)) {
+      return false;
+    }
```

Comment:
This should be removed and only used in credential-w3c
```suggestion
    if (await isRevoked(credential, context)) {
      return false;
    }
```